### ISSUE TITLE
How To Restore Gitlab After Failed Upgrade?

### ISSUE THREAD

**Issue by verrol** [sameersbn/docker-gitlab#649]
My entire gitlab server is hosted.

I am using Sameersbn/docker-gitlab:8.2.3 though Synoloyg NAS. After upgrading their DSM to 6.0, gitlab filed to start. I decided to backup the gitlab directory (with the repositories, backups, etc.). Then I removed the Gitlab docker image and tried to reinstall. That failed too and is still failing.

I want to big up sameersbn/docker-gitlab:8.2.3 on a linux box with external mysql server and redis container. Oh, the mysql db for gitlab was deleted by Synology when I used the package manager to remove gitlab.

With the back up files, can i restore my gitlab to the way it was before using a clean setup on another server? 

I would greatly appreciate some urgent help on this matter. Developers are unable to commit and share.

Thanks.

.v

**Comment by mpbzh** [comment]
I had the same issue. Check if its your DB connection that fails and try setting the Synology IP address as the db server address instead of the 127. something that used to work before.

**Comment by verrol** [comment]
Hi @mpbzh,
MiraDb seems to be fine. 

I have so far been able to restore the projects and mysql data from a back up file in the gitlab directory. Unfortunately, that was from Jan. 21. I think Synology upgraded Gitlab from 7.9.3 to 8.2.3. And my backup stopped working. I don't even know I had enable automatic backup, but I am glad I had that. It was a process to get it restored, but it restored the projects, and other MySQL database stuff.

But I would like to just deploy Gitlab 8.2.3, and have it come up with the latest data stored in the data dir.

Thanks for you time.

.v

**Comment by verrol** [comment]
The fact that I have the backups and can restore them. Is pretty good. For the projected that existed when the backup was taken, developers just have to push and it will be fine. For projects that were created after, they will just have to recreate them.

.v

**Comment by sameersbn** [comment]
Not sure about your troubles with Synology nas. But if you had automated backups enabled then the complete gitlab instance would have been backed up on a periodic basis. The backups are saved in a backups directory of the `data/` volume. Simple perform a restore operation with the `8.2.3` gitlab image and select the most recent backup.

If you are moving the gitlab instance to a linux box, then you should also be able to upgrade to the latest gitlab version after the restore.

**Comment by patrickse** [comment]
I had the same problem. What I´ve did is, remove my gitlab container and recreate it through docker-compose. My data files where mounted as volumes, so no data loss

**Comment by sameersbn** [comment]
@verrol were you able to resolve the issue? If not try adding `DEBUG=1` to enable bash debugging. Hopefully it would point to the source of the issue.

**Comment by Blackn0va** [comment]
My Gitlab Server was stuck after update to MariaDB10, always says Paket starting..

My Solution:  Login to MariaDB as root, Drop database from Gitlab , Delete User Gitlab_user für MariaDB 5, and mariaDB10. After this restart your nas, maybe twice. Then you can go to Paket Store and make a fresh Gitlab installation.

In my case i hafe a Backup from My Repositorys, restore it, and Done.

**Comment by GertschiM** [comment]
Balckn0va
I have the same problem,
Question how did you do the backup? can you explain the steps to me?

**Comment by Blackn0va** [comment]
Sure: 

I loggedin on phpMyAdmin, create a backup from Gitlab Database, 
log in via SSH and Type "docker exec -t GitLab gitlab-rake gitlab:backup:create"

IN MARIADB 5 AND 10 !
Then in phpMyAdmin drop database GitLab

Then Restart your NAS, after this you can Remove Gitlab from PaketStation.

Restart NAS, after This you can try to reinstall GitLab.

**Comment by GertschiM** [comment]
Thx,
but i hat now a problem



I can only run a backup on a working system ;-)
root@WIFDNA0010:~# docker exec -t synology_gitlab gitlab-rake gitlab:backup:create
Error response from daemon: Container ae378e7e43667613ec8ae5afc7f5145ec1b0784bb35b8e231266af597fd87436 is not running

and the restore command is?



_____________________________________________________________________________________________________________________________________________
Gerhard Mack | Senior IT Systems Architect

Schindler Fahrtreppen International GmbH | FIVN
Wienerbergstrasse 21-25 | 1100 Wien, Austria
_____________________________________________________________________________________________________________________________________________

**Comment by Blackn0va** [comment]
The Restore command is 

sudo gitlab-rake gitlab:backup:restore BACKUP=1512477187_2017_12_05_9.4.4_gitlab_backup

**Comment by Blackn0va** [comment]
Okay, you right, i had a backup from my system..

But you can try to backup manually these files and Folders.
Database =  work with Broke sys
Repositorys = Work with broke sys
Artifacts = I think it works
builds = Work
lfs = I don´t know
Pages
Uploads

All those files you can find on  (in my case)
/volume1/docker/gitlab

in you case you can try to Backup all files, destroy the database and make a clean fresh install and restore your files.

**Comment by GertschiM** [comment]
Thx,
thanks, will test it
god thank a second synology, where I can test this

_____________________________________________________________________________________________________________________________________________
Gerhard Mack | Senior IT Systems Architect

Schindler Fahrtreppen International GmbH | FIVN
Wienerbergstrasse 21-25 | 1100 Wien, Austria
_____________________________________________________________________________________________________________________________________________

**Comment by Blackn0va** [comment]
i becomes in few minutes direkt support from synology, maybe then i can tell you a better solution.

**Comment by GertschiM** [comment]
thx

_____________________________________________________________________________________________________________________________________________
Gerhard Mack | Senior IT Systems Architect

Schindler Fahrtreppen International GmbH | FIVN
Wienerbergstrasse 21-25 | 1100 Wien, Austria
_____________________________________________________________________________________________________________________________________________

**Comment by Blackn0va** [comment]
i have a full backup from Gitlab but i try now something and give you a response

**Comment by Blackn0va** [comment]
1. Go to phpMyAdmin and make a backup from gitlab Database, 
2. Then login via SSH and make a backup from you comletely gitlab Folder 
3. cp -R /volume1/docker/gitlab /volume1/homes/Black/docker/gitlab.backup
4. Drop The Gitlab Database
5. Restart NAS
6. Try Remove  Gitlab via PaketStation
7. Restart NAS
8. Install GitLab from PaketStation USE THE SAME DATABASE ANME PORT AND PASSWORDS !!
9. Restore your Gilab Database Backup via phpMyAdmin
10. Restore gitlab Folder via SSH 
11. cp -R /volume1/homes/Black/docker/gitlab.backup /volume1/docker/gitlab 
12. Restart NAS and Try to Login

On my Working System it will work..

**Comment by GertschiM** [comment]
Thx,
I'll test it right away

_____________________________________________________________________________________________________________________________________________
Gerhard Mack | Senior IT Systems Architect

Schindler Fahrtreppen International GmbH | FIVN
Wienerbergstrasse 21-25 | 1100 Wien, Austria
_____________________________________________________________________________________________________________________________________________

**Comment by GertschiM** [comment]
but we are already talking about the MariaDB10
and from the last version


_____________________________________________________________________________________________________________________________________________
Gerhard Mack | Senior IT Systems Architect

Schindler Fahrtreppen International GmbH | FIVN
Wienerbergstrasse 21-25 | 1100 Wien, Austria
_____________________________________________________________________________________________________________________________________________

**Comment by Blackn0va** [comment]
Yes you make backups from mariaDB 5 and install the newest version on MariaDB10

**Comment by Blackn0va** [comment]
One more step you need to restore your gitlab data..
sudo cp 1493107454_2017_04_25_9.1.0_gitlab_backup.tar /volume1/docker/gitlab/backups

login ssh 
docker exec -it synology_gitlab bash
and restore your data
bundle exec rake gitlab:backup:restore RAILS_ENV=production
